src/gsIO/gsWriteParaview.hpp

```diff
@@ -1660,6 +1661,144 @@ void gsWriteParaviewTrimmedCurve(const gsTrimSurface<T>& surf,
+
+        gsVector<unsigned> np = uniformSampleCount(a,b,npts);
+        gsMatrix<> pts = gsPointGrid(a,b,np);
+
+        gsMatrix<> eval_geo;
+        geometry.eval_into(pts,eval_geo);
+        std::map<std::string, gsMatrix<> > data;
+        for (int i = 0; i < fields.nFields(); ++i)
+        {
+            fields.isParametric(i) ? fields.igaFunction(p,i).eval_into(pts,data[fields.name(i)])
+                                   : fields.function(p,i).eval_into(eval_geo,data[fields.name(i)]);
+
+            if (data[fields.name(i)].rows() == 2)
+            {
+                data[fields.name(i)].conservativeResize(3,eval_geo.cols() );
+                data[fields.name(i)].row(2).setZero();
+            }
+        }
+
+        if ( 3 - d > 0)
+        {
+            np.conservativeResize(3);
+            np.bottomRows(3-d).setOnes();
+        }
+        if ( 3 - n > 0 )
+        {
+            eval_geo.conservativeResize(3,eval_geo.cols() );
+            eval_geo.bottomRows(3-n).setZero();
+        }
+
+        gsWriteParaviewTPgrid(eval_geo, data, np.template cast<index_t>(), fn + util::to_string(p));
+
+        //=========================================================================//
+
+        collection.addPart(fileName + util::to_string(p),".vts");
+
+        if (mesh)
+        {
+            // if none of the fields is a BSpline function use basis of the geometry to plot mesh (may be coarse)
+            int index = -1;
+            for (int i = 0; i < fields.nFields(); ++i)
+                if (fields.isParametrized(i))
+                    index = i;
+            const gsBasis<T> & basis = (index == -1) ? fields.patch(p).basis()
+                                                     : fields.igaFunction(p,index).basis();
+            writeSingleCompMesh(basis,fields.patch(p),fn + util::to_string(p) + "_mesh",nptsMesh);
+            collectionMesh.addPart(fileName + util::to_string(p) + "_mesh",".vtp");
+        }
+    }
+
+    collection.save();
+    if (mesh)
+        collectionMesh.save();
+}
+
+template<class T>
+void gsWriteParaviewTPgrid(gsMatrix<T> const& points,
+                           std::map<std::string, gsMatrix<T> > const& data,
+                           const gsVector<index_t> & np,
+                           std::string const & fn)
+{
+    const int n = points.rows();
+
+    std::string mfn(fn);
+    mfn.append(".vts");
+    std::ofstream file(mfn.c_str());
+    file << std::fixed; // no exponents
+    file << std::setprecision (PLOT_PRECISION);
+
+    file <<"<?xml version=\"1.0\"?>\n";
+    file <<"<VTKFile type=\"StructuredGrid\" version=\"0.1\">\n";
+    file <<"<StructuredGrid WholeExtent=\"0 "<< np(0)-1<<" 0 "<<np(1)-1<<" 0 "
+         << (np.size()>2 ? np(2)-1 : 0) <<"\">\n";
+    file <<"<Piece Extent=\"0 "<< np(0)-1<<" 0 "<<np(1)-1<<" 0 "
+         << (np.size()>2 ? np(2)-1 : 0) <<"\">\n";
+
+    file <<"<PointData>\n";
+    for (auto it = data.begin(); it != data.end(); ++it)
```

Comment:
I think `typename std::map<std::string, gsMatrix<T> >::const_iterator`

Reply:
Got it. Changed all instances of auto to an explicit pointer declaration